### PR TITLE
release: support universal macosx binary

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# What ❔

Add support for universal MacOS binary (x64 + arm64). For now, it is added additionally to existing arm64 and amd64 binaries.

## Tests

[Testing workflow](https://github.com/matter-labs/era-compiler-solidity/actions/runs/9741681761)

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
Could you outline some motivation why do we need this in addition to what @FlashSheridan proposed?

Our users don't migrate these binaries via Migration Assistant, but breaking Hardhat is unacceptable.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -117,6 +119,20 @@ jobs:
           pattern: release*
           path: releases
 
+      - name: Prepare universal binary
+        if: github.ref_type == 'tag' || (inputs.release_macos_amd64 && inputs.release_macos_arm64)
+        env:
+          MACOSX_UNIVERSAL_SUFFIX: "macosx-universal"
```
Comment (hedgar2017) [inline .github/workflows/release.yaml:125]: Could we remove the `universal` suffix?
Unless it is a widely-accepted and idiomatic naming for MacOS apps.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (antonbaliasnikov) [reply .github/workflows/release.yaml]: Removed.

**Comment by antonbaliasnikov** [comment]
> Please keep these binaries separately and do not include them into any of the release workflows. We'll have to figure this out with plugin devs later.

@hedgar2017 , I removed the suffix, tested, and disabled generation for now. As soon as hardhat will be ready, we can enable it.

**Comment by hedgar2017** [comment]
> > Please keep these binaries separately and do not include them into any of the release workflows. We'll have to figure this out with plugin devs later.
> 
> 
> 
> @hedgar2017 , I removed the suffix, tested, and disabled generation for now. As soon as hardhat will be ready, we can enable it.

We can probably start releasing them, if we remove the target suffix altogether.
Or we can be releasing them instead of one of the existing targets, given that the other target will stay intact.
Wdyt?

**Comment by antonbaliasnikov** [comment]
> > > Please keep these binaries separately and do not include them into any of the release workflows. We'll have to figure this out with plugin devs later.
> > 
> > 
> > @hedgar2017 , I removed the suffix, tested, and disabled generation for now. As soon as hardhat will be ready, we can enable it.
> 
> We can probably start releasing them, if we remove the target suffix altogether. Or we can be releasing them instead of one of the existing targets, given that the other target will stay intact. Wdyt?

I would suggest releasing it as an additional binary first, so, we'll have:
* `zksolc-macosx-arm64-v1.5.x`
* `zksolc-macosx-amd64-v1.5.x`
* `zksolc-macosx-v1.5.x` -> universal one

Then, we can check with hardhat plugin guys what is required to switch to an universal binary, and after some period when the switch is successful, deprecate arm64 and amd64 ones.

**Comment by hedgar2017** [comment]
> > > > Please keep these binaries separately and do not include them into any of the release workflows. We'll have to figure this out with plugin devs later.
> > > 
> > > 
> > > @hedgar2017 , I removed the suffix, tested, and disabled generation for now. As soon as hardhat will be ready, we can enable it.
> > 
> > 
> > We can probably start releasing them, if we remove the target suffix altogether. Or we can be releasing them instead of one of the existing targets, given that the other target will stay intact. Wdyt?
> 
> I would suggest releasing it as an additional binary first, so, we'll have:
> 
> * `zksolc-macosx-arm64-v1.5.x`
> * `zksolc-macosx-amd64-v1.5.x`
> * `zksolc-macosx-v1.5.x` -> universal one
> 
> Then, we can check with hardhat plugin guys what is required to switch to an universal binary, and after some period when the switch is successful, deprecate arm64 and amd64 ones.

Nice! Let's do it for now.

**Review by hedgar2017 — APPROVED** [review]
Thanks!
Please make sure to update zksolc- and zkvyper-bin as well!

**Pull request merged**